lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -254,6 +305,70 @@ describe('BasicLtiLaunchApp', () => {
     });
   });
 
+  describe('multiple fetches', () => {
+    let contentUrlCall;
+    let groupsCall;
+
+    beforeEach(() => {
+      // Will attempt to fetch the 1. content url and 2. groups.
+      fakeConfig.api = {
+        authToken: 'dummyAuthToken',
+        viaCallbackUrl: 'https://lms.hypothes.is/api/files/1234',
+        sync: {
+          data: {
+            course: {
+              context_id: '12345',
+              custom_canvas_course_id: '101',
+            },
+          },
+          path: '/api/sync',
+        },
+      };
+      contentUrlCall = fakeApiCall.withArgs({
+        authToken: 'dummyAuthToken',
+        path: 'https://lms.hypothes.is/api/files/1234',
+      });
+
+      groupsCall = fakeApiCall.withArgs({
+        authToken: 'dummyAuthToken',
+        path: '/api/sync',
+        data: {
+          course: {
+            context_id: '12345',
+            custom_canvas_course_id: '101',
+          },
+        },
+      });
+    });
+    it('renders the spinner until both groups and contentUrl requests finish', async () => {
+      const contentUrl = contentUrlCall.resolves({
+        via_url: 'https://via.hypothes.is/123',
+      });
+      const groups = groupsCall.resolves(['group1', 'group2']);
+      const wrapper = renderLtiLaunchApp();
+      await contentUrl;
+      // Spinner should not go away after first request
+      assert.isTrue(wrapper.exists('Spinner'));
+
+      await groups;
+      wrapper.update();
+      // Spinner should hide after second request finishes
+      assert.isFalse(wrapper.exists('Spinner'));
+    });
+    it('shows an error dialog if the first request fails and second succeeds', async () => {
```

Comment:
What about the case where the first request succeeds and the second one fails?

Reply:
I didn't bother because its the same result as a single request just failing, but I'll add that test in any case.